### PR TITLE
Add pwe support for sap

### PR DESCRIPTION
**Pull request by Rex-Chen-NXP**
1. sap support sae pwe
2. wifi status support all sae pwe types

### CODE REVIEW

File: modules/hostap/src/supp_api.c [hunk 1]
```diff
@@ -1499,22 +1520,31 @@ int supplicant_status(const struct device *dev, struct wifi_iface_status *status
...
 		int proto;
 		int key_mgmt;
+		int sae_pwe;
 
 		if (!ssid) {
 			wpa_printf(MSG_ERROR, "Failed to get current ssid");
 			goto out;
 		}
 
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
```
Comment (MaochenWang1) [inline modules/hostap/src/supp_api.c:1535]: `supplicant_status` is only used for NON-hostapd case, why need to `CONFIG_WIFI_NM_HOSTAPD_AP` here?
If `CONFIG_WIFI_NM_HOSTAPD_AP` is undefined, then `sae_pwe` is uninitialized.

Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: updated as comment.

**Review by MaochenWang1 — DISMISSED** [review]

File: modules/hostap/src/supp_api.c [hunk 2]
```diff
@@ -967,8 +973,8 @@ static int wpas_add_and_config_network(struct wpa_supplicant *wpa_s,
 			    params->security == WIFI_SECURITY_TYPE_SAE_AUTO) {
 				if (!wpa_cli_cmd_v("set sae_pwe %d",
 						   (params->security == WIFI_SECURITY_TYPE_SAE_H2E)
-							   ? 1
-							   : 2)) {
+						   ? 1
+						   : 2)) {
```
Comment (krish2718) [inline modules/hostap/src/supp_api.c:977]: please remove whitespace changes

File: include/zephyr/net/wifi.h [hunk 3]
```diff
@@ -87,6 +89,12 @@ enum wifi_security_type {
 	/** FT-EAP-SHA384 security */
 	WIFI_SECURITY_TYPE_FT_EAP_SHA384,
 
+	/** WPA/WPA2/WPA3 PSK security, sae with hash-to-element. */
+	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL_H2E,
+	/** WPA/WPA2/WPA3 PSK security, sae with both hunting-and-pecking
+	 *  loop and hash-to-element enabled.
+	 */
+	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL_AUTO,
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:97]: This is a bit confusing, seeing two AUTOs :). May be drop this and reuse `WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL` to mean everything is auto?

File: modules/hostap/src/supp_api.c (hunk 2 above)
Comment (Rex-Chen-NXP) [reply modules/hostap/src/supp_api.c]: Okay, updated as comment.

File: include/zephyr/net/wifi.h [hunk 4]
```diff
@@ -66,6 +66,8 @@ enum wifi_security_type {
 	WIFI_SECURITY_TYPE_WPA_PSK,
 	/** WPA/WPA2/WPA3 PSK security. */
 	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL,
+	/** WPA/WPA2/WPA3 PSK security, sae with hunting-and-pecking loop. */
```
[the remark this answers is not in the document]
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h]: updated all as comment.

File: include/zephyr/net/wifi.h (hunk 3 above)
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h]: Yes, of course, it should be like this, actually I notice this point early.
updated as comment.

File: include/zephyr/net/wifi.h [hunk 5]
```diff
@@ -86,6 +88,10 @@ enum wifi_security_type {
 	WIFI_SECURITY_TYPE_FT_EAP,
 	/** FT-EAP-SHA384 security */
 	WIFI_SECURITY_TYPE_FT_EAP_SHA384,
+	/** WPA/WPA2/WPA3 PSK security, SAE with hunting-and-pecking loop. */
+	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL_HNP,
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:92]: Can you please update commit log as to why this was necessary? Is there a specific reason or issue this solves?

**Review by krish2718 — DISMISSED** [review]

**Review by MaochenWang1 — APPROVED** [review]

File: include/zephyr/net/wifi.h (hunk 5 above)
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h:92]: For mix mode connection, need set pwe type for SAE to compatible with ext-AP, will connect failed if not compatible.

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h:92]: Updated commit info.

Comment (krish2718) [reply include/zephyr/net/wifi.h:92]: but in AUTO mode we do enable both HnP and H2E, so, that should work with any AP config, no?

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h:92]: Yes, but AUTO mode mapping to mix mode with sae pwe type 0 if the impl without this change.

**Comment by Rex-Chen-NXP** [comment]
Hi @jukkar 
As CI check failed, I rebased code, could you help review?

File: include/zephyr/net/wifi.h (hunk 5 above)
Comment (krish2718) [reply include/zephyr/net/wifi.h:92]: https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hostap/src/supp_api.c#L2330 it's 2.

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h:92]: yes, sorry my mistake, what I want to express is the original impl just cover only one sae pwe type.

Comment (krish2718) [reply include/zephyr/net/wifi.h:92]: `2` mean both HnP and H2E. Can you please share the problem that this PR is solving?

**Review by MaochenWang1 — CHANGES_REQUESTED** [review]
Need more check about mix mode change

File: include/zephyr/net/wifi.h (hunk 5 above)
Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h:92]: Yes, I got your point, I tried to get history of the issue related to this PR, but only get info that if not configure pwe as 2 for mix mode, connect failed with AP include 0x32 IE that specify bss require H2E.

1. Without this change, I test mix mode with AP include 0x32 IE, connect success.
2. And mix mode should compatible all sae pwe type, so the current impl(mix mode set pwe as 2) is suitable.

I think should remove the change about mix mode of this PR. What do you think about it? @krish2718 
Avoid PR merged, maochen help to revert the approve.

Comment (krish2718) [reply include/zephyr/net/wifi.h:92]: Yes, IMHO dropping https://github.com/zephyrproject-rtos/zephyr/pull/82956/commits/3d57b012c2c1f2ad31612cfea4fb7a84977b4ac0 should be fine, remaingin two commits are for SAP.

Comment (Rex-Chen-NXP) [reply include/zephyr/net/wifi.h]: Updated as comment.

**Review by krish2718 — APPROVED** [review]

**Review by MaochenWang1 — APPROVED** [review]

**Comment by Rex-Chen-NXP** [comment]
Approved by two reviewers almost 1 week ago. Could help to merge this PR?

**Comment by MaochenWang1** [comment]
Hi @kartben please help review this PR.
We have coming commits that depend on this PR and want to move forward.

**Review by jukkar — APPROVED** [review]

**Comment by kartben** [comment]
> Hi @kartben please help review this PR.
> We have coming commits that depend on this PR and want to move forward.

You needed approval from the assignee (and now you have it :))

**Pull request merged**